### ISSUE TITLE
[BUG/ISSUE] When i run the co2 simulation, got In NcGet_Var_Attr_C: cannot read attribute : units

### ISSUE THREAD

**Issue by hgrhgy** [geoschem/geos-chem#328]
##  Envirenment
GC VERSION: 12.8.0
##  Configuration
rundirs created by UT_12.8.0
- CopyRunDirs.input
```shell
# !INPUTS:
#
# %%% ID tags %%%
#
   VERSION        : 12.8.0
   DESCRIPTION    : Create run directory from UnitTest
#
# %%% Data path and HEMCO settings %%%
#
   GCGRID_ROOT    : /share/nas1_share2/ExtData
   DATA_ROOT      : {GCGRIDROOT}/
   VERBOSE        : 0
   WARNINGS       : 1
#
# %%% Code and queue settings %%%
#
   CODE_DIR       : {HOME}/geoschem/GC/Code.12.8.0
#
# %%% Unit tester path names %%%
#
   UNIT_TEST_ROOT : {HOME}/geoschem/UT/UT.12.8.0
   RUN_ROOT       : {UTROOT}/runs
   RUN_DIR        : {RUNROOT}/{RUNDIR}
   PERL_DIR       : {UTROOT}/perl
#
# %%% Target directory and copy command %%%
# 
   COPY_PATH      : {HOME}/geoschem/GC/rundirs
   COPY_CMD       : cp -rfL

#
#--------|-----------|------|----------------|------------|--------------|-----|
# MET    | GRID      | NEST | SIMULATION     | START DATE | END DATE     |EXTRA|
#--------|-----------|------|----------------|------------|--------------|-----|
## ======= Carbon Dioxide =====================================================
 geosfp   2x25        -      CO2              2017030100   2017040100     -
# merra2   2x25        -      CO2              2016070100   2016080100     -

!END OF RUNS:
#EOP
#------------------------------------------------------------------------------

```

## Problem
i run the co2 simulation, and got the error "In NcGet_Var_Attr_C: cannot read attribute : units". It seems the file 'HEMCO/GFED4/v2020-02/2017/GFED4_dailyfrac_gen.025x025.201703.nc' not contains the attr 'units'.  re-download the file, and the problen not solved.

- HEMCO.log in rundirs
```

Reading fields of HEMCO configuration file: HEMCO_Config.rc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/BIO/Net_terrestrial_exch_5.29Pg.generic.1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/TIMEZONES/v2015-02/timezones_voronoi_1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//GEOS_2x2.5/GEOS_FP/2011/01/GEOSFP.20110101.CN.2x25.nc
HEMCO: Opening ./GEOSChem.Restart.20170301_0000z.nc4
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/OLSON_MAP/v2019-02/Olson_2001_Land_Type_Masks.025x025.generic.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/FOSSIL/TIMES_weekly_scale_factors.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/FOSSIL/Aviation_SurfCorr_SclFac.1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/FOSSIL/ODIAC_CO2.monthly.generic.1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/OCEAN/Scaled_Ocean_CO2_monthly.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CEDS/v2018-08/2014/CO2-em-anthro_CMIP_CEDS_2014.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/AEIC/v2015-01/AEIC.47L.gen.1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/FOSSIL/ODIAC_CO2.monthly.generic.1x1.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/CHEM/CH4_source.geos.2x25.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/CHEM/Isoprene-2004.geos.2x25.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2015-04/CHEM/Monoterpene-2004.geos.2x25.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/GFED4/v2020-02/2017/GFED4_gen.025x025.201703.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/CO2/v2019-02/CHEM/CO2_prod_rates.GEOS5.2x25.47L.nc
HEMCO: Opening /share/nas1_share2/ExtData//HEMCO/GFED4/v2020-02/2017/GFED4_dailyfrac_gen.025x025.201703.nc

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

In NcGet_Var_Attr_C: cannot read attribute : units

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Code stopped from DO_ERR_OUT (in module NcdfUtil/m_do_err_out.F90) 

This is an error that was encountered in one of the netCDF I/O modules,
which indicates an error in writing to or reading from a netCDF file!

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

```

- ncdump the file
```sh
(geoschem) huangyj@wm02:~/geoschem$ ncdump -cts /share/nas1_share2/ExtData//HEMCO/GFED4/v2020-02/2017/GFED4_dailyfrac_gen.025x025.201703.nc
netcdf GFED4_dailyfrac_gen.025x025.201703 {
dimensions:
	time = 31 ;
	lat = 720 ;
	lon = 1440 ;
variables:
	int64 time(time) ;
		time:units = "days since 2017-03-01 00:00:00" ;
		time:calendar = "proleptic_gregorian" ;
		time:_Storage = "chunked" ;
		time:_ChunkSizes = 31 ;
		time:_DeflateLevel = 1 ;
		time:_Endianness = "little" ;
	float lat(lat) ;
		lat:_FillValue = NaNf ;
		lat:standard_name = "latitude" ;
		lat:long_name = "latitude" ;
		lat:units = "degrees_north" ;
		lat:axis = "Y" ;
		lat:_Storage = "chunked" ;
		lat:_ChunkSizes = 720 ;
		lat:_DeflateLevel = 1 ;
		lat:_Endianness = "little" ;
	float lon(lon) ;
		lon:_FillValue = NaNf ;
		lon:standard_name = "longitude" ;
		lon:long_name = "longitude" ;
		lon:units = "degrees_east" ;
		lon:axis = "X" ;
		lon:_Storage = "chunked" ;
		lon:_ChunkSizes = 1440 ;
		lon:_DeflateLevel = 1 ;
		lon:_Endianness = "little" ;
	float GFED_FRACDAY(time, lat, lon) ;
		GFED_FRACDAY:_FillValue = NaNf ;
		GFED_FRACDAY:_Storage = "chunked" ;
		GFED_FRACDAY:_ChunkSizes = 8, 180, 360 ;
		GFED_FRACDAY:_DeflateLevel = 1 ;
		GFED_FRACDAY:_Endianness = "little" ;

// global attributes:
		:_NCProperties = "version=1|netcdflibversion=4.6.1|hdf5libversion=1.10.2" ;
		:_SuperblockVersion = 2 ;
		:_IsNetcdf4 = 1 ;
		:_Format = "netCDF-4" ;
data:

 time = "2017-03-01", "2017-03-02", "2017-03-03", "2017-03-04", "2017-03-05", 
    "2017-03-06", "2017-03-07", "2017-03-08", "2017-03-09", "2017-03-10", 
    "2017-03-11", "2017-03-12", "2017-03-13", "2017-03-14", "2017-03-15", 
    "2017-03-16", "2017-03-17", "2017-03-18", "2017-03-19", "2017-03-20", 
    "2017-03-21", "2017-03-22", "2017-03-23", "2017-03-24", "2017-03-25", 
    "2017-03-26", "2017-03-27", "2017-03-28", "2017-03-29", "2017-03-30", 
    "2017-03-31" ;

 lat = -89.875, -89.625, -89.375, -89.125, -88.875, -88.625, -88.375, ...
    89.875 ;

 lon = -179.875, -179.625, -179.375, -179.125, -178.875, -178.625, -178.375,  ...
    177.625, 177.875, 178.125, 178.375, 178.625, 178.875, 179.125, 179.375, 
    179.625, 179.875 ;
}
```

**Comment by yantosca** [comment]
Thanks for bringing this fix to our attention.  These fields were provided by a third party, and evidently the `GFED4_3hrfrac_gen_*.nc` and `GFED4_dailyfrac_gen*.nc` files lacked a unit attribute.  These are all fractions so the units should be "1" (aka unitless).

I have used NCO to add a units attribute to these files:
```
HEMCO/v2020-02/2017/GFED4_3hrfrac_gen.*.nc
HEMCO/v2020-02/2017/GFED4_dailyfrac_gen.*.nc
HEMCO/v2020-02/2018/GFED4_3hrfrac_gen.*.nc
HEMCO/v2020-02/2018/GFED4_dailyfrac_gen.*.nc
HEMCO/v2020-02/2019/GFED4_3hrfrac_gen.*.nc
HEMCO/v2020-02/2019/GFED4_dailyfrac_gen.*.nc
```
@Jun-Meng, can you copy these files from the Harvard gcgrid to the ComputeCanada server?  Thanks much.

**Comment by yantosca** [comment]
@hgrhgy : You should be able to download these new files from:

- http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/GFED4/v2020-02/2017
- http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/GFED4/v2020-02/2018
- http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/GFED4/v2020-02/2019

for the time being until they are posted to ComputeCanada.

**Comment by Jun-Meng** [comment]
Hi Bob, 

I didn't see any change for those files. Are the modified files in another directory?  

Jun

**Comment by yantosca** [comment]
Oh, our FTP has a delay before posting.  Try again later today.

**Comment by Jun-Meng** [comment]
right on. Just copied.

**Comment by yantosca** [comment]
Thanks Jun.  I'll close out this issue.  @hgrhgy, this problem should be resolved once you copy over the new data files.

**Comment by tianjialiu** [comment]
> Thanks for bringing this fix to our attention. These fields were provided by a third party, and evidently the `GFED4_3hrfrac_gen_*.nc` and `GFED4_dailyfrac_gen*.nc` files lacked a unit attribute. These are all fractions so the units should be "1" (aka unitless).

Hi Bob,
The maximum value of these files isn't 1. It seems like the daily fractions are multiplied by the number of days in the month, and the 3-hourly fractions by 8. Maybe there should be a note about this in the wiki or HEMCO_Config.rc to clarify the units? I created some new emissions files based on the HEMCO GFED format and just assumed the daily fractions meant a range of 0-1 like in the original dataset. And then, I found that this is probably the reason my model outputs were really off.
